examples/increment-counter/src/main.rs

```diff
@@ -9,117 +9,35 @@
-    // "0x20202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020202020666F6F"
-
-    // binary representation of the key used to store the counter value
-    let storage_key: StorageKey = [
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32, 32,
-        102, 111, 111,
-    ];
-    let mut buffer = [0u8; READ_BUFFER_LEN];
-
-    unsafe {
-        // Read from storage
-        match get(storage_key.as_ptr() as u32, buffer.as_mut_ptr() as u32) {
-            0 => (),
-            err => return err,
-        }
-
-        // Decode counter structure
-        let mut counter = if buffer == [0u8; READ_BUFFER_LEN] {
-            // Default value if zero
-            Counter::default()
-        } else {
-            Counter::decode(&mut &buffer[..]).unwrap_or(Counter::default())
-        };
-
-        // Increment by value
-        let new_value = counter.counter.saturating_add(increment);
-        call_print(&format!(
-            "incrementing counter: {} + {} -> {}",
-            counter.counter, increment, new_value
-        ));
-        counter.counter = new_value;
-
-        // Encode counter structure
-        buffer.fill(0);
-        for (pos, elem) in Counter::encode(&counter).iter().enumerate() {
-            buffer[pos] = *elem
-        }
+#[no_mangle]
+#[polkavm_derive::polkavm_export]
+pub extern "C" fn call() {
+    // Accept increment value from user input. Should be 4 bytes (e.g., `0x12345678`) or `ContractTrapped` error occurs.
+    input!(increment: u32, );
```

Comment:
Do we have an option to use SCALE encoding for input data?

Related to: https://github.com/QuantumFusion-network/qf-polkavm-sdk/pull/20#discussion_r2118008875

Reply:
As far as I understand `input` already expected a SCALE encoded value. For instance if I provide `0xffffff0` - SCALE encoded unsigned 32-bit integer 16777215 according to https://docs.polkadot.com/polkadot-protocol/parachain-basics/data-encoding/#data-types - to the input field in the Polkadot/Substrate Portal (see image below) it generates the event with `Counter updated from 542520506 to 559297721.` message (second image below) where `559297721 - 542520506 = 16777215`.

<img width="1714" height="785" alt="image" src="https://github.com/user-attachments/assets/d0b985e7-d60b-4b73-baa8-7ad0d677a28c" />

<img width="842" height="201" alt="image" src="https://github.com/user-attachments/assets/12460c19-aea2-4214-967b-faaedfc4c11d" />